src/Sulu/Component/Content/Compat/Serializer/PageBridgeHandler.php

```diff
@@ -91,16 +121,18 @@ public function doDeserialize(
         array $type,
         Context $context
     ) {
-        $document = $context->accept($data['document'], ['name' => PageDocument::class]);
-
+        $document = $context->accept($data['document'], ['name' => $data['documentClass']]);
         $structure = $this->structureFactory->getStructureMetadata('page', $data['structure']);
-
         $bridge = new PageBridge($structure, $this->inspector, $this->propertyFactory, $document);
 
         // filthy hack to set the Visitor::$result to null and force the
         // serializer to return the Bridge and not the Document
         $visitor->setNavigator($context->getNavigator());
 
+        if (!$this->registry->hasDocument($document)) {
+            $this->registry->registerDocument($document, $this->nodeManager->find($data['uuid']));
+        }
```

Comment:
hmm not really because if you deserialize a document which was serialized in the same session it would be placed twice into the registry and if you deserialize a document from a file which which was created in a different session you have to add it to the registry...